### PR TITLE
codecov: use -fkeep-inline-functions for native_posix code coverage

### PR DESCRIPTION
**Pull request by dcpleung**
Add GCC options -fkeep-inline-functions for native_posix to be used
for code coverage. This asks GCC to still keep the body of these
inline functions around so gcov can properly record execution
count of such functions (according to gcov document).

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

### CODE REVIEW

File: kernel/thread.c [hunk 1]
```diff
@@ -261,7 +261,6 @@ void z_check_stack_sentinel(void)
 }
 #endif
 
-#ifdef CONFIG_MULTITHREADING
```
Comment (aescolar) [inline kernel/thread.c]: Is this change related?

Comment (dcpleung) [reply kernel/thread.c]: Yes. Or else the linker would complain about undefined references.

Though, this is just a test to see how to get gcov/codegen to cover those inline functions (hence the draft status). I need to look into the report in details to see if these compiler options actually help or make things worse.

**Comment by dcpleung** [comment]
Turning this on makes CONFIG_MULTITHREADING to have no effect, even though I tried to revert the kconfig changes and exclude these functions in the headers, resulting in segfault for the no-multithreading test. So don't use this option.
